presto-main/src/main/java/com/facebook/presto/geospatial/H3HexFunctions.java

```diff
@@ -0,0 +1,367 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.geospatial;
+
+import com.facebook.presto.common.block.ArrayBlockBuilder;
+import com.facebook.presto.common.block.Block;
+import com.facebook.presto.common.block.BlockBuilder;
+import com.facebook.presto.common.type.StandardTypes;
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.function.Description;
+import com.facebook.presto.spi.function.ScalarFunction;
+import com.facebook.presto.spi.function.SqlNullable;
+import com.facebook.presto.spi.function.SqlType;
+import com.uber.h3core.AreaUnit;
+import com.uber.h3core.H3CoreV3;
+import com.uber.h3core.LengthUnit;
+import com.uber.h3core.util.LatLng;
+import io.airlift.slice.Slice;
+import io.airlift.slice.Slices;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.facebook.presto.common.type.DoubleType.DOUBLE;
+import static com.facebook.presto.common.type.VarcharType.VARCHAR;
+import static com.facebook.presto.spi.StandardErrorCode.INVALID_FUNCTION_ARGUMENT;
+
+public final class H3HexFunctions
+{
+    private static final H3CoreV3 h3;
+
+    private H3HexFunctions() {}
+
+    static {
+        try {
+            h3 = H3CoreV3.newInstance();
+        }
+        catch (IOException ex) {
+            throw new RuntimeException("Failed to load H3Core", ex);
+        }
+    }
+
+    private static Block convertHexagonAddrArrayToBlock(List<String> hexAddrs)
+    {
+        BlockBuilder block = VARCHAR.createBlockBuilder(null, hexAddrs.size());
+        for (String hexAddr : hexAddrs) {
+            VARCHAR.writeString(block, hexAddr);
+        }
+        return block.build();
+    }
+
+    private static List<String> convertBlockToHexagonAddrs(Block hexAddrBlock)
+    {
+        List<String> hexAddrsStr = new ArrayList<>();
+        for (int i = 0; i < hexAddrBlock.getPositionCount(); i++) {
```

Comment:
```suggestion
        for (int i = 0, ct = hexAddrBlock.getPositionCount(); i < ct; i++) {
```
old fashion, not relying on the compiler's optimization but doing it manually :-)